client/jmx-adapter/src/test/java/org/jolokia/client/jmxadapter/JmxBridgeTest.java

```diff
@@ -0,0 +1,765 @@
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.createMBean("java.lang.Object", RUNTIME, RUNTIME);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.createMBean("java.lang.Object", RUNTIME, new Object[0], new String[0]);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.createMBean("java.lang.Object", RUNTIME, RUNTIME, new Object[0], new String[0]);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.unregisterMBean(RUNTIME);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.addNotificationListener(RUNTIME, (NotificationListener) null, null, null);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.addNotificationListener(RUNTIME, RUNTIME, null, null);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.removeNotificationListener(RUNTIME, RUNTIME);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.removeNotificationListener(RUNTIME, RUNTIME, null, null);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.removeNotificationListener(RUNTIME, (NotificationListener) null);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+    try {
+      this.adapter.removeNotificationListener(RUNTIME, (NotificationListener) null, null, null);
+      Assert.fail("Operation should not be supported by adapter");
+    } catch (UnsupportedOperationException ignore) {
+    }
+  }
+
+  @Test
+  public void testOverallOperations() throws IOException {
+    final MBeanServerConnection nativeServer = getNativeConnection();
+    Assert.assertEquals(
+        this.adapter.getMBeanCount(),
+        nativeServer.getMBeanCount(),
+        "Number of MBeans are the same");
+
+    Assert.assertEqualsNoOrder(
+        this.adapter.getDomains(),
+        nativeServer.getDomains(),
+        "Domain list is the same");
+
+    Assert.assertEquals(
+        this.adapter.getDefaultDomain(), nativeServer.getDefaultDomain(), "Default domain");
+
+    Assert.assertEquals(this.adapter.agentVersion, "1.6.2");
```

Comment:
@skarsaune I think this will give us some headaches when releasing. We should either remove this test or make it at parameterised with a property from the pom.xml